spark-etl/src/main/scala/geotrellis/spark/etl/Etl.scala

```diff
@@ -72,35 +80,92 @@ case class Etl(args: Seq[String], @transient modules: Seq[TypedModule] = Etl.def
 
-  def reproject[
-  K: SpatialComponent: ClassTag,
-  V <: CellGrid: ClassTag: Stitcher: ? => TileReprojectMethods[V]: ? => CropMethods[V]: ? => TileMergeMethods[V]: ? => TilePrototypeMethods[V]
-  ](rdd: RDD[(K, V)] with Metadata[M]): (Int, RDD[(K, V)] with Metadata[M]) = {
-    val tuple = rdd.reproject(conf.crs(), conf.layoutScheme()(conf.crs(), conf.tileSize()))
-    tuple._2.persist(conf.cache())
-    tuple
-  }
+  /**
+    * Tiles RDD of arbitrary rasters to conform to a layout scheme or definition provided in the arguments.
+    * First metadata will be collected over input rasters to determine the overall extent, common crs, and resolution.
+    * This information will be used to select a LayoutDefinition if LayoutScheme is provided in the arguments.
+    *
+    * The tiling step will use this LayoutDefinition to cut input rasters into chunks that conform to the layout.
+    * If multiple rasters contribute to single target tile their values will be merged cell by cell.
+    *
+    * The timing of the reproject steps depends on the method chosen.
+    * BufferedReproject must be performed after the tiling step because it leans on SpatialComponent to identify neighboring
+    * tiles and sample their edge pixels. This method is the default and produces the best results.
+    *
+    * PerTileReproject method will be performed before the tiling step, on source tiles. When using this method the
+    * reproject logic does not have access to pixels past the tile boundary and will see them as NODATA.
+    * However, this approach does not require all source tiles to share a projection.
+
+    *
+    * @param rdd    RDD of source rasters
+    * @param method Resampling method to be used when merging raster chunks in tiling step
+    */
+  def tile[
+    I: ProjectedExtentComponent: (? => TilerKeyMethods[I, K]),
+    V <: CellGrid: Stitcher: ClassTag: (? => TileMergeMethods[V]): (? => TilePrototypeMethods[V]):
+    (? => TileReprojectMethods[V]): (? => CropMethods[V]),
+    K: SpatialComponent: Boundable: ClassTag
+  ](
+    rdd: RDD[(I, V)], method: ResampleMethod = NearestNeighbor
+  )(implicit sc: SparkContext): (Int, RDD[(K, V)] with Metadata[RasterMetaData]) = {
+    val targetCellType = conf.cellType.get
+    val destCrs = conf.crs()
+
+    val source = // reproject source tiles before performing any mosaicing
+      if (conf.reproject() == PerTileReproject) {
+        rdd.reproject(destCrs)
+      } else {
+        rdd
+      }
+
+    val (tiledZoom: Int, rmd: RasterMetaData) = {
+      scheme match {
+        case Left(layoutScheme) =>
+          RasterMetaData.fromRdd(source, layoutScheme)
+        case Right(layoutDefinition) =>
+          RasterMetaData.fromRdd(source, layoutDefinition)
+      }
+    }
+    val adjustedMetadata = targetCellType.fold(rmd){ ct => rmd.copy(cellType = ct) }
+    val tiled: RDD[(K, V)] with Metadata[RasterMetaData] =
+      ContextRDD(source.tileToLayout[K](adjustedMetadata, method), adjustedMetadata)
```

Comment:
This isn't right. If we are doing a buffered reproject, then we always tile to a floating layout scheme using the source tile CRS. *Then* we reproject and tile to the desired layout. If we reproject against the source tiles, this will work, but for the buffered tile case, the first tileToLayout is wrong.